### PR TITLE
Explore making state part of the find data

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Making the change to put state in RawFindData as suggested by @svick on https://github.com/dotnet/designs/pull/24#issuecomment-348785224. I think this way is better- note that I haven't added state to the transform delegate before, but it is intended to have it per the spec.

@svick, @danmosemsft, @terrajobst

### CODE REVIEW

File: src/System.IO.FileSystem/src/System/IO/FindPredicate.cs [hunk 1]
```diff
@@ -7,5 +7,5 @@ namespace System.IO
     /// <summary>
     /// Interface for filtering out find results.
     /// </summary>
-    internal delegate bool FindPredicate<TState>(ref RawFindData findData, TState state);
+    internal delegate bool FindPredicate<TState>(ref RawFindData<TState> findData);
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/FindPredicate.cs]: If both the delegate types have the same signature now, how about just defining a delegate:
```C#
internal delegate bool FindFunc<TState>(ref RawFindData<TState> findData);
```
and using the same delegate for both?

File: src/System.IO.FileSystem/src/System/IO/FindTransform.cs [hunk 2]
```diff
@@ -7,5 +7,5 @@ namespace System.IO
     /// <summary>
     /// Delegate for transforming raw find data into a result.
     /// </summary>
-    internal delegate T FindTransform<T>(ref RawFindData findData);
+    internal delegate TResult FindTransform<TResult, TState>(ref RawFindData<TState> findData);
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/FindTransform.cs]: Generally the result type comes last.  But see my earlier comment about unifying the delegates, anyway.

**Comment by karelz** [comment]
@JeremyKuhne looks like stale PR. Any plans to push it forward?

**Comment by karelz** [comment]
@JeremyKuhne ping?

**Comment by JeremyKuhne** [comment]
This is out of date.

**Pull request closed without merging**